### PR TITLE
Fixed a number of missing initialisations detected by Eclipse.

### PR DESCRIPTION
**Pull request by heplesser**
This PR fixes a number of missing initialisations that Eclipse alerted me to (in combination with clang). In a few cases, comments are inserted marking the warning as not relevant.

### CODE REVIEW

File: models/aeif_psc_delta.cpp [hunk 1]
```diff
@@ -271,7 +273,9 @@ nest::aeif_psc_delta::State_::set( const DictionaryDatum& d, const Parameters_&,
   updateValueParam< double >( d, names::w, y_[ W ], node );
 }
 
-nest::aeif_psc_delta::Buffers_::Buffers_( aeif_psc_delta& n )
+
+nest::aeif_psc_delta::Buffers_::Buffers_( // @suppress("Class members should be properly initialized")
```
Comment (clinssen) [inline models/aeif_psc_delta.cpp:277]: Is this ``// @suppress`` Eclipse syntax? Is it desirable to have these comments in upstream?

This also raises the question why the directive is here (what are we suppressing and why)?

Comment (heplesser) [reply models/aeif_psc_delta.cpp]: Fixed, I removed this again, it is too Eclipse-specific. The warning is about missing initialisation, but that is handled in `init_buffers()`, and this again explained in a constructor comment.

File: nestkernel/sparse_node_array.h [hunk 2]
```diff
@@ -60,7 +60,11 @@ class SparseNodeArray
 public:
   struct NodeEntry
   {
-    NodeEntry() = default;
+    NodeEntry()
+      : node_( nullptr )
+      , node_id_( 0 )
+    {
+    }
```
Comment (hakonsbm) [inline nestkernel/sparse_node_array.h:67]: Same here, the defaulted constructor will zero-initialize the variables.

**Review by hakonsbm — CHANGES_REQUESTED** [review]

**Review by clinssen — APPROVED** [review]

File: libnestutil/block_vector.h [hunk 3]
```diff
@@ -79,7 +79,11 @@ class bv_iterator
   using reference = ref_;
   using difference_type = typename BlockVector< value_type >::difference_type;
 
-  bv_iterator() = default;
+  bv_iterator()
+    : block_vector_( nullptr )
+    , block_index_( 0 )
+  {
+  }
```
[the remark this answers is not in the document]
Comment (heplesser) [reply libnestutil/block_vector.h:86]: But based on the Python mantra of "explicit is better than implicit" and to make Eclipse happy it does not hurt, does it? So I would like to keep it as is.

File: nestkernel/sparse_node_array.h (hunk 2 above)
Comment (heplesser) [reply nestkernel/sparse_node_array.h:67]: See above.

File: libnestutil/block_vector.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (hakonsbm) [reply libnestutil/block_vector.h:86]: Fair enough.

**Review by hakonsbm — APPROVED** [review]

**Pull request merged**